src/averageAges.js

```diff
@@ -37,7 +48,21 @@ function calculateMenAverageAge(people, century) {
  * @return {number}
  */
 function calculateWomenAverageAge(people, withChildren) {
-  // write code here
+  let totalAge = 0;
+  let count = 0;
+
+  people.forEach(person => {
+    if (person.sex === 'f' && (!withChildren || hasChildren(person, people))) {
+      totalAge += person.died - person.born;
+      count++;
+    }
+  });
+
+  return count === 0 ? 0 : totalAge / count;
```

Comment:
Instead of checking if count equals 0 at the end, check at the start if the array is empty and if it is, return 0.